### PR TITLE
prefer 'assert' to 'assert_eq' for bools

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: capnp/src/private/layout_test.rs [hunk 1]
```diff
@@ -61,26 +61,26 @@ fn simple_raw_data_struct() {
...
-        assert_eq!(reader.get_bool_field(8),  true);
-        assert_eq!(reader.get_bool_field(9),  true);
-        assert_eq!(reader.get_bool_field(10), false);
-        assert_eq!(reader.get_bool_field(11), false);
-        assert_eq!(reader.get_bool_field(12), false);
-        assert_eq!(reader.get_bool_field(13), true);
-        assert_eq!(reader.get_bool_field(14), false);
-        assert_eq!(reader.get_bool_field(15), false);
+        assert!(reader.get_bool_field(0));
+        assert!(!reader.get_bool_field(1));
```
Comment (dwrensha) [inline capnp/src/private/layout_test.rs:65]: The `assert_eq!()` version is much more clear for this.

Comment (dwrensha) [reply capnp/src/private/layout_test.rs:65]: At a glance, it's easy to overlook the `!`.

Comment (danieleades) [reply capnp/src/private/layout_test.rs:65]: noted. I'll close this PR, and this lint can be suppressed

**Comment by danieleades** [comment]
closing. this lint will be suppressed

**Pull request closed without merging**